tool/movement.re

```diff
@@ -0,0 +1,138 @@
+                Text { text: "Flying" },
+            ),
+            selected: movement_mode.number,
+            onchange: fn(index: int) { movement_mode.number = index; },
+        }),
+        UiElement::Button(UiButton {
+            label: Text { text: "Set Movement Mode" },
+            onclick: fn(label: Text) { Tas::set_movement_mode(movement_mode.number); },
+        }),
+        UiElement::Input(Input {
+            label: FLYING_UP_DOWN_VELOCITY_LABEL,
+            input: "300",
+            onclick: fn(input: string) {
+                let val = input.parse_float();
+                match val {
+                    Result::Ok(val) => { MOVEMENT_STATE.fly_down_up_velocity = val; },
+                    Result::Err(e) => { FLYING_UP_DOWN_VELOCITY_LABEL.text = "Up/Down Flying Velocity [error: invalid input]"; },
+                }
+            },
+            onchange: fn(input: string) {},
+        }),
+        UiElement::Input(Input {
+            label: FLYING_FORWARD_BACKWARD_VELOCITY_LABEL,
+            input: "600",
+            onclick: fn(input: string) {
+                let val = input.parse_float();
+                match val {
+                    Result::Ok(val) => { Tas::set_forward_backward_fly_speed(val); },
+                    Result::Err(e) => { FLYING_FORWARD_BACKWARD_VELOCITY_LABEL.text = "Forward/Backward Flying Velocity [error: invalid input]"; },
+                }
+            },
+            onchange: fn(input: string) {},
+        }),
+        UiElement::Button(UiButton {
+            label: Text { text: "Back" },
+            onclick: fn(label: Text) { leave_ui() },
+        }),
+    ))
+}
+
+static MOVEMENT_COMPONENT = Component {
+    id: MOVEMENT_COMPONENT_ID,
+    conflicts_with: List::of(MOVEMENT_COMPONENT_ID),
+    draw_hud: fn(text: string) -> string { text },
+    tick_mode: TickMode::DontCare,
+    requested_delta_time: Option::None,
+    on_tick: fn() {
+        if MOVEMENT_STATE.enable_fly {
+            Tas::set_movement_mode(5);
+            if MOVEMENT_STATE.is_flying_up && !MOVEMENT_STATE.is_flying_down {
+                if Tas::get_movement_mode() == 5 {
+                    let vel = Tas::get_velocity();
+                    Tas::set_velocity(Velocity { x: vel.x, y: vel.y, z: MOVEMENT_STATE.fly_down_up_velocity });
+                }
+            } else if !MOVEMENT_STATE.is_flying_up && MOVEMENT_STATE.is_flying_down {
+                if Tas::get_movement_mode() == 5 {
+                    let vel = Tas::get_velocity();
+                    Tas::set_velocity(Velocity { x: vel.x, y: vel.y, z: (MOVEMENT_STATE.fly_down_up_velocity * -1.) });
+                }
+            } else if !MOVEMENT_STATE.is_flying_up && !MOVEMENT_STATE.is_flying_down {
+                if Tas::get_movement_mode() == 5 {
+                    let vel = Tas::get_velocity();
+                    Tas::set_velocity(Velocity { x: vel.x, y: vel.y, z: 0. });
+                }
+            }
+        }
```

Comment:
You should use an enum for the three different cases. It more precisely allows you to express your intent. It also allows you to use `match` on the enum, making the code cleaner. Also you ensure that you are always only ever in one of the three states (currently, what happens when both `is_flying_up` and `is_flying_down` is `true`?).